### DISCUSSION TITLE
Having issue with adding this integration

### DISCUSSION THREAD

**Discussion by Babolater** [astrandb/miele#451] [Q&A]
I'm attempting to install this integration, but keep running into the same issue. I can get through the Miele pages. After being asked in HA to enter the Client ID and Client Secret, get redirected to a Miele page where it asks to log into Miele's services. I log in and get directed to the page that ask if I want to link to Home Assistant. I select "Link Account" and then it kicks me back to HA, where I receive an error message that says "OAuth authorization error while obtaining access token." (snip of the error below). Not sure how to progress past this.

<img width="420" alt="Screenshot 2024-03-16 at 11 52 55 AM" src="https://github.com/astrandb/miele/assets/155329791/81356b92-c338-47e1-a2b0-444a8c0b105d">

**Comment by metawops** [comment]
Same problem here. 😞

**Reply by Babolater** [reply]
This ended up being a me issue or resolved itself with the passage of time. I had taken a screenshot of the Client ID/Client Secret and was copy/pasting that information from the screenshot to the HA integration. Well when I went back to try again several weeks later I noticed that when pasting the information one of the zeros/“O” characters was not copying correctly so the information that I was putting into HA was incorrect. I also noticed that you could go into your account on the Miele app and see the third-party integrations and if the integration is already in there and the system isn’t working you might try removing the integration and starting the set up process over again.
